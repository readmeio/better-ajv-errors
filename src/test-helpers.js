import { readFileSync } from 'fs';
import path from 'path';

export async function getSchemaAndData(name, dirPath) {
  const schemaPath = path.join(dirPath, '..', '__fixtures__', name, 'schema.json');
  const schema = JSON.parse(readFileSync(schemaPath, 'utf8'));
  const dataPath = path.join(dirPath, '..', '__fixtures__', name, 'data.json');
  const json = readFileSync(dataPath, 'utf8');
  const data = JSON.parse(json);

  return [schema, data, json];
}
