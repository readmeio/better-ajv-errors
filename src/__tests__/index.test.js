import Ajv from 'ajv';
import { getSchemaAndData } from '../test-helpers';
import betterAjvErrors from '..';
import betterAjvErrorsBabelExport from '../../lib';

describe('Main', () => {
  it('should output error with reconstructed codeframe', async () => {
    const [schema, data] = await getSchemaAndData('default', __dirname);
    const ajv = new Ajv();
    const validate = ajv.compile(schema);
    const valid = validate(data);
    expect(valid).toBe(false);

    const res = betterAjvErrors(schema, data, validate.errors, {
      format: 'cli',
      indent: 2,
    });
    expect(res).toMatchSnapshot();
  });

  it('should output error with codeframe', async () => {
    const [schema, data, json] = await getSchemaAndData('default', __dirname);
    const ajv = new Ajv();
    const validate = ajv.compile(schema);
    const valid = validate(data);
    expect(valid).toBe(false);

    const res = betterAjvErrors(schema, data, validate.errors, {
      format: 'cli',
      json,
    });
    expect(res).toMatchSnapshot();
  });

  describe('Babel export', () => {
    it('should function as normal', async () => {
      const [schema, data] = await getSchemaAndData('default', __dirname);
      const ajv = new Ajv();
      const validate = ajv.compile(schema);
      const valid = validate(data);
      expect(valid).toBe(false);

      const res = betterAjvErrorsBabelExport(schema, data, validate.errors);
      expect(res).toMatchSnapshot();
    });
  });
});
