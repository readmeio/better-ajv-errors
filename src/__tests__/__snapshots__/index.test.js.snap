// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Babel export should function as normal 1`] = `
"[31m[1mENUM[22m[39m[31m must be equal to one of the allowed values[39m
[31m(paragraph, codeBlock, blockquote)[39m

> 1 | {"type":"doc","version":1,"content":[{"type":"paragarph"}]}
    |                                              ^^^^^^^^^^^ 👈🏽  Did you mean [95mparagraph[39m here?"
`;

exports[`Main complex schema examples should output an error on an invalid OpenAPI 3.1 definition 1`] = `
"[31m[1mUNEVALUATED PROPERTY[22m[39m[31m must NOT have unevaluated properties[39m

  15 |         "summary": "Example operation",
  16 |         "description": "This operation has \`tags\` typod.",
> 17 |         "tagss": [
     |         ^^^^^^^ 😲  [95mtagss[39m is not expected to be here!
  18 |           "Tag Name"
  19 |         ],
  20 |         "responses": {"
`;

exports[`Main should output error with codeframe 1`] = `
"[31m[1mENUM[22m[39m[31m must be equal to one of the allowed values[39m
[31m(paragraph, codeBlock, blockquote)[39m

  2 |   "type": "doc",
  3 |   "version": 1,
> 4 |   "content": [{ "type": "paragarph" }]
    |                         ^^^^^^^^^^^ 👈🏽  Did you mean [95mparagraph[39m here?
  5 | }
  6 |"
`;

exports[`Main should output error with reconstructed codeframe [without colors] 1`] = `
"ENUM must be equal to one of the allowed values
(paragraph, codeBlock, blockquote)

  4 |   "content": [
  5 |     {
> 6 |       "type": "paragarph"
    |               ^^^^^^^^^^^ 👈🏽  Did you mean paragraph here?
  7 |     }
  8 |   ]
  9 | }"
`;

exports[`Main should output error with reconstructed codeframe 1`] = `
"[31m[1mENUM[22m[39m[31m must be equal to one of the allowed values[39m
[31m(paragraph, codeBlock, blockquote)[39m

  4 |   "content": [
  5 |     {
> 6 |       "type": "paragarph"
    |               ^^^^^^^^^^^ 👈🏽  Did you mean [95mparagraph[39m here?
  7 |     }
  8 |   ]
  9 | }"
`;
